### PR TITLE
PICARD-1402: Register macOS app for supported file types

### PR DESCRIPTION
**Pull request by phw**
# Summary
Allows opening supported file types on macOS from Finder or by drag and drop on the dock icon.

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-1402

# Solution
Register for supported file types. Register for actually supported types (see https://developer.apple.com/library/archive/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html) and also by file extension (needed for filetypes otherwise unknown to macOS, e.g. Musepack).

# Action

### CODE REVIEW

File: picard/tagger.py [hunk 1]
```diff
@@ -391,7 +391,11 @@ def event(self, event):
         if isinstance(event, thread.ProxyToMainEvent):
             event.run()
         elif event.type() == QtCore.QEvent.FileOpen:
-            self.add_files([event.file()])
+            file = event.file()
+            if os.path.isdir(file):
+                self.add_directory(file)
+            else:
+                self.add_files([event.file()])
```
[the remark this answers is not in the document]
Comment (phw) [reply picard/tagger.py]: This adds support for directly dropping folder onto the app icon, see https://github.com/metabrainz/picard/pull/1432/commits/f7dd0cada967234ee87da2171f4ed77469aab427. The type `public.folder` is added to the Info.plist there. But this only works if our FileOpen handler also handles directories.

Also see how this specifies the UTI `public.folder`. This is a subtype of the also defined `public.directory`. The difference is that `public.folder` is what users would usually consider a folder (a directory with a bunch of files in it the user can access). But some special directories are not `public.folder`. E.g. a macOS app is technically also only a directory with a specific structure inside (which the user in normal usage does not access directly). By defining `public.folder` as a supported type users can drag and drop their folders containing the music files into Picard, but not e.g. an app.

File: picard/plugin.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def unregister_module(self, name):
             pass
 
     def __iter__(self):
-        enabled_plugins = config.setting["enabled_plugins"]
+        enabled_plugins = config.setting["enabled_plugins"] if config.setting else []
```
[the remark this answers is not in the document]
Comment (phw) [reply picard/plugin.py]: In picard.spec `picard.formats.supported_formats` gets imported. But there is no config in this case and `config.setting` is not available.

Checking for this here allows us to use `supported_formats` to get the standard formats supported by Picard programmatically, but we don't care about plugin support here. Of course the downside is that any format supported by plugins is not set as a supported file type. But file type support is actually mostly statically defined on all OSes anyway.

File: picard/tagger.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zas) [reply picard/tagger.py]: Ok, thanks for the explanation.
BTW, is there a reason to not write:
```python
self.add_files([file])
```

[the remark this answers is not in the document]
Comment (phw) [reply picard/tagger.py]: Nope, just bad copy and paste. I'll fix it

**Pull request merged**